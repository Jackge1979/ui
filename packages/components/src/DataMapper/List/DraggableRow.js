import React, { Component } from 'react';
import { DragSource, DropTarget } from 'react-dnd';
import PropTypes from 'prop-types';
import flow from 'lodash/flow';
import Row from './Row.js';
import { ItemTypes } from '../Constants';

const elementSource = {
	beginDrag(props) {
		return props.dndListener.beginDrag(props.element);
	},
	endDrag(props) {
		props.dndListener.endDrag();
	},
};

const elementTarget = {
	drop(props, monitor) {
		props.dndListener.drop(monitor.getItem(), props.element);
	},
	canDrop(props, monitor) {
		const sourceItem = monitor.getItem();
		return props.dndListener.canDrop(sourceItem, props.element);
	},
};

function collectForDragSource(connect, monitor) {
	return {
		connectDragSource: connect.dragSource(),
		isDragging: monitor.isDragging(),
	};
}

function collectForDropTarget(connect, monitor) {
	return {
		connectDropTarget: connect.dropTarget(),
		dragOver: monitor.isOver(),
	};
}

/**
 * This component adds the draggable behaviour to the Row component.
 */
class DraggableRow extends Component {

	render() {
		const {
			element,
			classNameProvider,
			dataKeys,
			onClick,
			onDoubleClick,
			rowDataGetter,
			rowRenderers,
			onEnterElement,
			onLeaveElement,
			connectDragSource,
			connectDropTarget,
		} = this.props;
		return connectDragSource(
			connectDropTarget(
				<div>
					<Row
						element={element}
						onClick={onClick}
						onDoubleClick={onDoubleClick}
						classNameProvider={classNameProvider}
						dataKeys={dataKeys}
						rowDataGetter={rowDataGetter}
						rowRenderers={rowRenderers}
						onEnterElement={onEnterElement}
						onLeaveElement={onLeaveElement}
					/>
				</div>,
			),
		);
	}
}

DraggableRow.propTypes = {
	element: PropTypes.object,
	classNameProvider: PropTypes.object,
	dataKeys: PropTypes.array,
	rowDataGetter: PropTypes.object,
	rowRenderers: PropTypes.object,
	onClick: PropTypes.func,
	onDoubleClick: PropTypes.func,
	onEnterElement: PropTypes.func,
	onLeaveElement: PropTypes.func,
	connectDragSource: PropTypes.func,
	connectDropTarget: PropTypes.func,
	dragOver: PropTypes.bool,
};

export default flow(
	DragSource(ItemTypes.ELEMENT, elementSource, collectForDragSource),
	DropTarget(ItemTypes.ELEMENT, elementTarget, collectForDropTarget),
)(DraggableRow);
