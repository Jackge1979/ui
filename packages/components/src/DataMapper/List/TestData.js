import RowLabel from './RowLabel';
import MandatoryField from './MandatoryField';

export const element1 = {
	id: 'elem_1',
	name: 'element 1',
	type: 'string',
	description: 'bla bla bla',
  mandatory: false,
};

export const element2 = {
	id: 'elem_2',
	name: 'element 2',
	type: 'string',
	description: 'This element is mandatory',
  mandatory: true,
};

export const KEYS = {
  NAME: 'name',
  TYPE: 'type',
  DESC: 'desc',
};

export const dataKeys = [KEYS.NAME, KEYS.TYPE, KEYS.DESC];

export const classNameProvider = {
  get(element, key) {
    if (element && key) {
			return `classname-of-(${element.id},${key})`;
		} else if (element) {
			return `classname-of-(${element.id})`;
		}
		return `classname-of-list`;
  },
  onDragOver(status) {
  },
};

export const rowDataGetter = {
  getId(element) {
    return element.id;
  },
  getData(element, key) {
		switch (key) {
			case KEYS.NAME:
				return {
					value: element.name,
					mandatory: element.mandatory,
				};
			case KEYS.TYPE:
				return element.type;
			case KEYS.DESC:
				return element.description;
			default:
				return `No data available for ${key}`;
		}
	},
};

export const rowRenderers = {
  getComponent(key) {
		switch (key) {
			case KEYS.NAME:
				return MandatoryField;
			default:
				return RowLabel;
		}
	},
};
