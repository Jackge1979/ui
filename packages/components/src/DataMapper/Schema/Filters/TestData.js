export const Filters = {
	WITH: true,
	WITHOUT: false,
};

export const element1 = {
	id: 'elem_1',
	name: 'Firstname',
	type: 'string',
	description: 'bla bla bla',
	mandatory: false,
};

export const element2 = {
	id: 'elem_2',
	name: 'Lastname',
	type: 'string',
	description: 'bla bla bla',
	mandatory: true,
};

export const element3 = {
	id: 'elem_3',
	name: 'Birthday',
	type: 'date',
	description: 'bla bla bla',
	mandatory: false,
};

export const element4 = {
	id: 'elem_4',
	name: 'Address',
	type: 'address',
	description: 'bla bla bla',
	mandatory: true,
};

export const schema = {
	id: '5df4g6ds4df564s',
	name: 'schema',
	elements: [element1, element2, element3, element4],
};
