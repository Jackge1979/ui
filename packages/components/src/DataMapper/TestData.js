
export const element1 = {
	id: 'elem_1',
	name: 'Firstname',
	type: 'string',
	description: 'The firstname of the customer (optional)',
	mandatory: false,
};

export const element2 = {
	id: 'elem_2',
	name: 'Lastname',
	type: 'string',
	description: 'The lastname of the customer (mandatory)',
	mandatory: true,
};

export const element3 = {
	id: 'elem_3',
	name: 'Birthday',
	type: 'date',
	description: 'Thje birthday of the customer (optional)',
	mandatory: false,
};

export const element4 = {
	id: 'elem_4',
	name: 'Address',
	type: 'address',
	description: 'The address of the customer (mandatory)',
	mandatory: true,
};

export const schema = {
	id: '5df4g6ds4df564s',
	name: 'Customer info',
	elements: [element1, element2, element3, element4],
};
