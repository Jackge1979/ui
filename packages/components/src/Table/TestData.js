// This module defines data, objects and functions used by the unit tests.

export const element1 = {
	id: 'elem_1',
	name: 'element 1',
	type: 'string',
	desc: 'Description of element 1',
};

export const element2 = {
	id: 'elem_2',
	name: 'element 2',
	type: 'string',
	desc: 'This element is a string',
};

export const element3 = {
	id: 'elem_3',
	name: 'Firstname',
	type: 'string',
	desc: 'The firstname of the customer (optional)',
	mandatory: false,
};

export const element4 = {
	id: 'elem_4',
	name: 'Lastname',
	type: 'string',
	desc: 'The lastname of the customer (mandatory)',
	mandatory: true,
};

export const element5 = {
	id: 'elem_5',
	name: 'Birthday',
	type: 'date',
	desc: 'The birthday of the customer (optional)',
	mandatory: false,
};

export const element6 = {
	id: 'elem_6',
	name: 'Address',
	type: 'address',
	desc: 'The address of the customer (mandatory)',
	mandatory: true,
};

export const Columns = {
	NAME: {
		key: 'name',
		label: 'Name',
	},
	TYPE: {
		key: 'type',
		label: 'Type',
	},
	DESC: {
		key: 'desc',
		label: 'Description',
	},
	MANDATORY: {
		key: 'mandatory',
		label: '',
	},
};

export const columns1 = [Columns.NAME, Columns.TYPE, Columns.DESC];

function getColumn(col) {
	return {
		key: col.key,
		label: col.label,
		headClassName: `classname-of-head-${col.key}`,
		cellClassName: `classname-of-row-data-${col.key}`,
	};
}

export const columns2 = [getColumn(Columns.NAME), getColumn(Columns.TYPE), getColumn(Columns.DESC)];

export const classnames = {
	table: 'classname-of-table',
	header: 'classname-of-table-header',
	body: 'classname-of-table-body',
	rows: ['classname-of-row-1', 'classname-of-row-2'],
};

export const rowDataGetter = {
	getElementId(element) {
		return element.id;
	},
	getRowData(element, columnKey) {
		switch (columnKey) {
			case Columns.NAME.key:
				return element.name;
			case Columns.TYPE.key:
				return element.type;
			case Columns.DESC.key:
				return element.desc;
			case Columns.MANDATORY.key:
				return element.mandatory ? '*' : '';
			default:
				return `No data available for ${columnKey}`;
		}
	},
};
