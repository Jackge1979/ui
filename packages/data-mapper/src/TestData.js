export const emptySchema = {
	id: 'empty_schema',
	name: 'Empty schema',
	elements: [],
};

export const element1 = {
	id: 'elem_1',
	name: 'Firstname',
	type: 'string',
	description: 'The firstname of the customer (optional)',
	mandatory: false,
};

export const element2 = {
	id: 'elem_2',
	name: 'Lastname',
	type: 'string',
	description: 'The lastname of the customer (mandatory)',
	mandatory: true,
};

export const element3 = {
	id: 'elem_3',
	name: 'Birthday',
	type: 'date',
	description: 'The birthday of the customer (optional)',
	mandatory: false,
};

export const element4 = {
	id: 'elem_4',
	name: 'Address',
	type: 'address',
	description: 'The address of the customer (mandatory)',
	mandatory: true,
};

export const schema1 = {
	id: '5df4g6ds4df564s',
	name: 'Customer info',
	elements: [element1, element2, element3, element4],
};

export const element5 = {
	id: 'elem_5',
	name: 'Name',
	type: 'string',
	description: 'The name of the client',
	mandatory: true,
};

export const element6 = {
	id: 'elem_6',
	name: 'Role',
	type: 'string',
	description: 'The role of the client',
	mandatory: false,
};

export const element7 = {
	id: 'elem_7',
	name: 'Birthday',
	type: 'date',
	description: 'The birthday of the client',
	mandatory: false,
};

export const element8 = {
	id: 'elem_8',
	name: 'Address',
	type: 'address',
	description: 'The address of the client',
	mandatory: true,
};

export const schema2 = {
	id: '1bf1dbw1d81w51',
	name: 'Client data',
	elements: [element5, element6, element7, element8],
};

export const mapping = [
	{
		source: element2,
		target: element5,
	},
	{
		source: element4,
		target: element8,
	},
];

export const Columns = {
	NAME: {
		id: 'name-col-id',
		key: 'name',
		label: 'Name',
	},
	TYPE: {
		id: 'type-col-id',
		key: 'type',
		label: 'Type',
	},
	DESC: {
		id: 'desc-col-id',
		key: 'description',
		label: 'Description',
	},
	MANDATORY: {
		id: 'mand-col-id',
		key: 'mandatory',
		label: '',
	},
};
