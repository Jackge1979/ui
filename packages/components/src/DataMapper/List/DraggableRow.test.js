import React from 'react';
import renderer from 'react-test-renderer';
import * as TestData from './TestData';
import DraggableRow from './DraggableRow';

it('single-row', () => {
	// Obtain the reference to the component before React DnD wrapping
	const OriginalRow = DraggableRow.DecoratedComponent;
	// Stub the React DnD connector functions with an identity function
	const identity = el => el;
	// create React tree
	const tree = renderer
		.create(
			<OriginalRow
				element={TestData.element1}
				classNameProvider={TestData.classNameProvider}
				dataKeys={TestData.dataKeys}
				rowDataGetter={TestData.rowDataGetter}
				rowRenderers={TestData.rowRenderers}
				connectDragSource={identity}
				connectDropTarget={identity}
			/>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
