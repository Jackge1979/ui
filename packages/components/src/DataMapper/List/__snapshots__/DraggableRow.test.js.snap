// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single-row 1`] = `
<div>
  <div
    className="comp-list-row classname-of-(elem_1,undefined)"
    data-id="elem_1"
    onClick={undefined}
    onDoubleClick={undefined}
  >
    <div
      className="comp-list-row-data mandatory-field classname-of-(elem_1,name)"
    >
      <div
        className="mandatory-field-label"
      >
        element 1
      </div>
      <div
        className="mandatory-field-info"
      >
        
      </div>
    </div>
    <div
      className="comp-list-row-data row-label classname-of-(elem_1,type)"
    >
      string
    </div>
    <div
      className="comp-list-row-data row-label classname-of-(elem_1,desc)"
    >
      bla bla bla
    </div>
  </div>
</div>
`;
